### PR TITLE
`Programming exercises`: Add more information to grading table

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

#5649 
Make the creation of test weights simpler, easier and less error-prone by providing more detail.

### Description

This PR adds some small things to make the grading table better:
- Each test now displays the amount of points it will grant for passing in the grading table.
- In the right column the total sum of all weights is now shown.
- The headers of the columns of the test case table are now also available in German.
- Since inactive test cases are already greyed out, the column for activity is not necessary and it did not provide any opportunity to sort the rows anyway. Therefore it is removed to save space.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis
2. Create a programming exercise
3. Go to the grading view
4. Change any values and check the shown values.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| programming-exercise-configure-grading.component.ts | 72% | 84% | 

### Screenshots

Bildschirmfoto 2022-09-26 um 01 58 04

### CODE REVIEW

**Review by ge65cer — COMMENTED** [review]
Tested on TS1, everything works. 👍 

On my screen, the table header is always abbreviated (datatable width is 804px). Could you maybe add a little more space, for example by allowing the collapse of the preview charts?
Screenshot 2022-09-27 at 16 02 15

**Review by simon-hng — COMMENTED** [review]
Code changes look good to me.
Also tested locally on my machine and it works beautifully.

Very nice responsive design with the breakpoint for medium screens!

File: src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.html [hunk 1]
```diff
@@ -98,7 +98,9 @@ <h2 class="mb-1 fw-medium" jhiTranslate="artemisApp.programmingExercise.configur
                             <ngx-datatable-column name="Test Name" prop="testName">
                                 <ng-template ngx-datatable-header-template let-sortFn="sortFn">
                                     <span class="datatable-header-cell-wrapper" (click)="sortFn()">
-                                        <span class="datatable-header-cell-label bold sortable">Test Name</span>
+                                        <span class="datatable-header-cell-label bold sortable">{{
+                                            'artemisApp.programmingExercise.configureGrading.gradingTableHeader.testName' | artemisTranslate
+                                        }}</span>
```
Comment (julian-christl) [inline src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.html:103]:
```suggestion
                                        <span class="datatable-header-cell-label bold sortable">
                                            {{'artemisApp.programmingExercise.configureGrading.gradingTableHeader.testName' | artemisTranslate}}
                                        </span>
```
This style would make more sense to me. Unless eslint complains. If you agree, please replace the other occurrences in your PR.

File: src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.html [hunk 2]
```diff
@@ -194,20 +205,38 @@ <h2 class="mb-1 fw-medium" jhiTranslate="artemisApp.programmingExercise.configur
...
-                                <ng-template ngx-datatable-cell-template let-value="value">
-                                    {{ value }}
+                                <ng-template ngx-datatable-cell-template let-row="row">
+                                    <input
+                                        class="table-editable-field__input form-control me-2"
+                                        [value]="testCasePoints[row.testName]"
+                                        type="text"
+                                        [disabled]="true"
+                                        [readOnly]="true"
+                                    />
```
Comment (julian-christl) [inline src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.html:231]: I don't see the sense behind adding an input that is always disabled. Then you can just write in a cell like the former "Is active" column or the "Type" column, right?

File: src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts [hunk 3]
```diff
@@ -589,11 +597,31 @@ export class ProgrammingExerciseConfigureGradingComponent implements OnInit, OnD
...
+     * Calculates the rounded points awarded for passing each test
+     */
+    updateTestPoints() {
+        this.testCasePoints = {};
+
+        if (!this.testCases) {
+            return;
+        }
+
+        const totalWeight = this.testCases.reduce((sum, testCase) => sum + testCase.weight!, 0);
```
Comment (julian-christl) [inline src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts:616]: I suggest replacing this with a global totalWeight value as in the charts.

Then in the `updateEditedField` recalculate the total weight value only if a weight field was edited. This way you only recalculate when necessary. Technically, if you have access to the value from before (not sure) you could even update the value without iterating over the whole array but I'm not sure if you do.

File: src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts [hunk 4]
```diff
@@ -589,11 +597,31 @@ export class ProgrammingExerciseConfigureGradingComponent implements OnInit, OnD
...
+        const maxPoints = this.programmingExercise.maxPoints!;
+        this.testCases.forEach((testCase) => {
+            if (testCase.testName) {
```
Comment (julian-christl) [inline src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts:619]: Why do we have this distinction? What is the use case here? Do we hide test cases without a name? I don't see an indication of that.

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 5]
```diff
@@ -199,6 +200,16 @@
                         "description": "Die Verteilung der Kategorien in Bezug auf 'Penalty', 'Issues' und 'Punkte'.<br/>Der <b>Penalty</b> Balken zeigt die Verteilung von Penalties in den Kategorien.<br/>Der <b>Issues</b> Balken zeigt wie viele Issues in der entsprechenden Kategorie erkannt wurden.<br/>Der <b>Punkte</b> Balken gibt an wie viele Punkte tatsächlich von den Ergebnissen abgezogen wurden durch eine Kategorie, was sich auch den vorherigen Balken ergibt.<br/>Bewege die Maus über einen farbigen Abschnitt, um mehr Details zu sehen."
                     }
                 },
+                "gradingTableHeader": {
+                    "testName": "Testname",
+                    "weight": "Gewichtung",
+                    "bonusMultiplier": "Bonus Multiplier",
```
Comment (julian-christl) [inline src/main/webapp/i18n/de/programmingExercise.json:206]:
```suggestion
                    "bonusMultiplier": "Bonus Multiplikator",
```

**Review by julian-christl — CHANGES_REQUESTED** [review]
Prevents approval:
- totalWeight recalculation (Reason: performance)

Should be addressed now or later:
- Input fields
- testName conditional
- Translation

Nice to have:
- {{...}} code style

**Review by julian-christl — APPROVED** [review]
Thank you for incorporating the changes.
Code approved.

**Review by canberkanar — DISMISSED** [review]
The added column LTM and the point calculation work correctly.

**Review by simon-hng — APPROVED** [review]
Still works as intended. Tested on TS1

**Review by julian-christl — APPROVED** [review]
Reapprove code

**Review by simon-hng — APPROVED** [review]
Second reapproval

**Review by ge65cer — APPROVED** [review]
Re-tested on TS1, no issues found. Code approved.

**Pull request merged**